### PR TITLE
ENH Change clear text to delete

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-linkfield/issues/149

Even though the ACs are to just change the text label, it also makes sense to change all references in the codebase from 'clear' to 'delete' so that we're not using two different word for the same thing

### CODE REVIEW

File: client/lang/en.js [hunk 1]
```diff
@@ -10,6 +10,6 @@ if (typeof(ss) === 'undefined' || typeof(ss.i18n) === 'undefined') {
     "LinkField.CONFIRM_DELETE": "Deleted link",
     "LinkField.DELETE_ERROR": "Failed to delete link",
     "LinkField.ADD_LINK": "Add Link",
-    "LinkField.CLEAR": "Clear"
+    "LinkField.Delete": "Delete"
```
Comment (ChloeSartorelli) [inline client/lang/en.js]:
```suggestion
    "LinkField.DELETE": "Delete"
```
I would normally ignore this, except that it's being referenced with all caps in the code so needs to be all caps here.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
Works well locally. Just one small change.

**Comment by GuySartorelli** [comment]
I just had a thought - and feel free to say it's out of scope or not worth doing. But should this say "Archive" when links are versioned, and "Delete" when they're not? That _is_ more accurate, but I'm on the fence as to whether it's valuable enough of a distinction to be worth adding the complexity.
(Note the functionality is already doing that, so it's just whether we dynamically change the wording or not)

**Comment by emteknetnz** [comment]
Have added in archived text functionality

**Review by GuySartorelli — APPROVED** [review]
Great! Works locally.

**Pull request merged**